### PR TITLE
update: thelounge configuration example

### PR DESCRIPTION
**Pull request by ishanjain28**

### CODE REVIEW

File: example_configs/thelounge.md [hunk 1]
```diff
@@ -5,12 +5,19 @@
...
-  tlsOptions: {},
-  primaryKey: "uid",
-  baseDN : "ou=people,dc=example,dc=com",
+      ldap: {
+        enable: true,
+        url: "ldap://localhost:389",
+        tlsOptions: {},
+        primaryKey: "uid",
+        searchDN: {
+            rootDN: "uid=ldap-editor,ou=people,DC=example,DC=com",
```
Comment (nitnelave) [inline example_configs/thelounge.md:14]: What's the root DN? And why ldap-editor? Maybe we need to describe this a bit more

Comment (ishanjain28) [reply example_configs/thelounge.md:14]: rootDN/rootPassword are the credentials used to login to lldap and perform queries. (Similar to bind user and bind user password. the lounge uses a different name for it) 

`ldap-editor` user is a member of `lldap_strict_readonly` and `lldap_password_manager` groups.

Comment (nitnelave) [reply example_configs/thelounge.md:14]: Right, this would be great to put in the documentation :)

**Comment by ishanjain28** [comment]
hey @nitnelave Please let me know if you have any more feedback on this PR. It is complete from my end and awaiting your approval

**Comment by nitnelave** [comment]
@ishanjain28  Sorry, my comment was stuck in "pending", I never actually sent it.

File: example_configs/thelounge.md [hunk 2]
```diff
@@ -5,12 +5,22 @@
...
+        // baseDN: "ou=people,dc=example,dc=com",
+        searchDN: {
+            rootDN: "uid=ldap-editor,ou=people,dc=example,dc=com",
+            rootPassword: ""
+            filter: "(memberOf=cn=thelounge,ou=groups,dc=example,dc=com)",
+            base: "dc=example,dc=com",
+            scope: "sub",
+        },
+    },
 ```
```
[the remark this answers is not in the document]
Comment (ishanjain28) [reply example_configs/thelounge.md:23]: That makes sense and I'll make this change.

**Comment by nitnelave** [comment]
Hey @ishanjain28 , do you want to update this PR, or should I take it over?

File: example_configs/thelounge.md [hunk 3]
```diff
@@ -2,15 +2,39 @@
 
 [Main documentation here.](https://thelounge.chat/docs/configuration#ldap-support)
 
-For simple user auth with LLDAP on localhost adapt this in the main config.js:
+Simple Config:
```
Comment (nitnelave) [inline example_configs/thelounge.md:5]: Make it a title, with 2 `#` before

File: example_configs/thelounge.md [hunk 4]
```diff
@@ -2,15 +2,39 @@
...
+        enable: true,
+        url: "ldap://localhost:389",
+        tlsOptions: {},
+        primaryKey: "uid",
+        baseDN: "ou=people,dc=example,dc=com",
+    },
 ```
 
-And comment out with ```//``` the entire ```searchDN``` block.
+In this config, The Lounge will use the credentials provided in web ui to authenticate with lldap. It'll allow access if authentication was successful and the user is a member of Base DN.
```
Comment (nitnelave) [inline example_configs/thelounge.md:17]: I would put the description first, before the config.

Also, all users are members of the base DN, so we don't need to specify that.

File: example_configs/thelounge.md [hunk 5]
```diff
@@ -2,15 +2,39 @@
...
+
+
+Advanced Config:
```
Comment (nitnelave) [inline example_configs/thelounge.md:20]: Title

File: example_configs/thelounge.md [hunk 6]
```diff
@@ -2,15 +2,41 @@
...
+            rootDN: "uid=ldap-viewer,ou=people,dc=example,dc=com",
+            rootPassword: ""
+            filter: "(memberOf=cn=thelounge,ou=groups,dc=example,dc=com)",
+            base: "dc=example,dc=com",
+            scope: "sub",
+        },
+    },
+```
+
+`rootDN` is similar to bind DN in other applications. It is used in combination with `rootPassword` to query lldap. `ldap-viewer` user in `lldap` is a member of the group `lldap_strict_readonly` group. This gives `ldap-viewer` user permission to query `lldap`.
```
Comment (nitnelave) [inline example_configs/thelounge.md:38]: Can you reset your password via the lounge? Also, the readonly permission is redundant, password manager is a superset.

**Review by nitnelave — CHANGES_REQUESTED** [review]

**Comment by ishanjain28** [comment]
Hey @nitnelave Apologies for the delays. 

I have updated the config into configs for simple and more advanced setups and rebased the PR. 

> The root DN is the bind user, preferably a member of strict readonly.

This was a great suggestion! I assumed the lounge supported changing passwords for users but they do not so we should not use a user with the permission to change passwords in this context, Just the user with read permissions is sufficient.

File: example_configs/thelounge.md (hunk 6 above)
Comment (ishanjain28) [reply example_configs/thelounge.md:38]: @nitnelave Yep, Updated this in the next commit

**Comment by ishanjain28** [comment]
I believe all of the comments have been addressed. I am not sure why github's UI is still stuck with "1 change requested"

**Review by nitnelave — APPROVED** [review]
Great, thanks!

**Comment by nitnelave** [comment]
It was just waiting for my approval :)

**Comment by nitnelave** [comment]
Thank you everyone for the contribution!
This feature is now available in the latest release, [v0.5.0](https://github.com/lldap/lldap/releases/tag/v0.5.0).
You can support LLDAP by starring our repo, contributing some configuration examples and becoming a sponsor.

**Pull request merged**